### PR TITLE
Issue 21136 refactor browser

### LINKED ISSUE #21283: SiteBrowser Portlet should use DB
This is part of a bigger issue: #21136 .

The idea here is to be able to load the SiteBrowser Portlet without indices and cache, so should be based on DB.

STR:
1- Start dotcms.
2- Navigate to Maintenance Portlet
3- Flush All Caches.
4- Deactivate both indices (live and working).
5- Navigate to Site Browser Portlet.

All content (file assets, folders and pages) should list up.

### PR DESCRIPTION
**Pull request by erickgonzalez**
Refactor BrowserQuery
Refactor to get the content/folders/links from DB instead of ES, at the Browser Portlet.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java [hunk 1]
```diff
@@ -223,5 +225,23 @@
...
     public Map<String, Object> getFolderContent(final BrowserQuery browserQuery) throws DotSecurityException, DotDataException;
-}
 
+
+	/**
+	 * Returns a collection of contentlets that live inside the parent(browserQuery.directParent)
+	 * @param browserQuery {@link BrowserQuery}
+	 * @return list of contentlets
+	 */
+	List<Contentlet> getContentUnderParentDB(BrowserQuery browserQuery);
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java:235]: I would rename to `getContentUnderParentFromDB`

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java [hunk 2]
```diff
@@ -52,55 +60,83 @@
...
-            if (host == null) {
+        try {
+            final List<Map<String,String>> inodesMapList =  dc.loadResults();
 
-                Logger.error(this, "Folder ID doesn't belong to a Folder nor a Host, id: " + browserQuery.hostFolderId
-                        + ", maybe the Folder was modified in the background.");
-                throw new NotFoundInDbException("Folder ID doesn't belong to a Folder nor a Host, id: " + browserQuery.hostFolderId);
+            final List<String> inodes = new ArrayList<>();
+            for (final Map<String, String> versionInfoMap : inodesMapList) {
+                final String inode = versionInfoMap.get("inode");
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java:84]: one line better?

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java [hunk 3]
```diff
@@ -314,35 +315,59 @@ private void includeLinks(final BrowserQuery browserQuery, final List<Map<String
...
+                    .getLinks((Folder) browserQuery.directParent, browserQuery.showWorking, browserQuery.showArchived,
+                            browserQuery.user,
+                            false);
+        }
+        return Collections.emptyList();
+    }
+
+
+
+    private  List<Map<String, Object>> includeFolders(final BrowserQuery browserQuery, final Role[] roles) throws DotDataException, DotSecurityException {
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java:343]: Doesn't look like a good method name

File: dotCMS/src/main/java/com/dotcms/browser/BrowserQuery.java [hunk 4]
```diff
@@ -1,144 +1,212 @@
...
+        this.showMenuItemsOnly = builder.showMenuItemsOnly;
+        this.host=hostAndFolder._1;
+        this.folder=hostAndFolder._2;
+        this.directParent = this.folder.isSystemFolder() ? host : folder;
+        this.roles= Try.of(()->APILocator.getRoleAPI().loadRolesForUser(user.getUserId()).toArray(new Role[0])).getOrElse(new Role[0]);
+
+    }
+
+
+    @CloseDBIfOpened
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserQuery.java:89]: no nice, but is ok

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java:235]: Why not use as paraneter directly the directParent attribute?

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java [hunk 5]
```diff
@@ -223,5 +225,23 @@
...
+
+	/**
+	 * Returns a collection of contentlets, folders, links that live inside the parent(browserQuery.directParent)
+	 * @param browserQuery {@link BrowserQuery}
+	 * @return list of treeable (folders, content, links)
+	 * @throws DotSecurityException
+	 * @throws DotDataException
+	 */
+	List<Treeable> getFolderContentList(BrowserQuery browserQuery) throws DotSecurityException, DotDataException;
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java:244]: same here

**Review by freddyucv — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**